apps/app/locals.tf

```diff
@@ -14,5 +14,14 @@
 
 locals {
   feature_store_odbc = "Driver={ODBC Driver 18 for SQL Server};Server=tcp:${data.azurerm_mssql_server.feature_store.fully_qualified_domain_name},1433;Database=${var.feature_store_db_name};Authentication=ActiveDirectoryMsi;Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;"
-  is_prod            = length(regexall("prod", lower(var.environment))) > 0
+  acr_repository     = var.app_id
+  create_repo        = var.app_config.managed_repo != null
```

Comment:
Thinking about the variable precedence, I think we'll need to have a block with the same name that get's overwritten by the environment-specific config, i.e. we change `managed_repo` to just `repo` then have a field for `create` that's either true or false?

Reply:
I'm not sure you could/should override anything about the repo in other than the first environment that gets deployed?

Reply:
Yeah this will be fine actually, as long as `managed_repo` isn't defined in the core `apps.yaml`, which we can stipulate in the docs. It should only be defined in the first environment, i.e. `apps.dev.yaml`, then left blank in `apps.test.yaml` and so on